### PR TITLE
chore: refactored rate-limit docs, added endpoint-based page and over…

### PR DESCRIPTION
**Pull request by tricky42**
## Checklist

- [ ] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md) and signed the CLA.
- [ ] I have referenced an issue containing the design document if my change introduces a new feature.
- [ ] I have read the [security policy](../security/policy).
- [ ] I confirm that this pull request does not address a security vulnerability. 
      If this pull request addresses a security vulnerability, 
      I confirm that I got approval (please contact [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push the changes.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added the necessary documentation within the code base (if appropriate).

### CODE REVIEW

**Review by vinckr — COMMENTED** [review]
If possible please dont create new documents, rather add more content to the existing document 
this keeps the sidebar readable and people dont drown in dozens of docs for slightly different topics. 
thanks!

File: docs/guides/rate-limits-overview.mdx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      per second, accommodating temporary spikes in traffic. <br/>**Sustained Limit:** This sets the maximum request count over a
+      one-minute period, ensuring consistent and fair usage.
+2. **Endpoint-Based Rate Limits** ([link](/guides/endpoint-based-rate-limits)): These limits focus on safeguarding specific
+   endpoints of your Ory projects against common attack vectors like brute-force and credential stuffing attempts. They typically
+   consider factors like source IP address and request frequency to identify and mitigate malicious activity.
+   1. Endpoint-based rate limits act as a first line of defense for your project endpoints.These limits often leverage the source
+      IP address as a key criterion, allowing Ory to identify and block requests originating from suspicious or malicious sources.
+   2. By analyzing request patterns and incorporating factors like source IP, Ory can effectively differentiate between legitimate
+      user traffic and potentially harmful bot activity.
+
```
Comment (Demonsthere) [inline docs/guides/rate-limits-overview.mdx:30]: I think this lacks certain clarification, as in the Project based rate limits we also distinguish based on certain paths/endpoints. 
Maybe we should rephrase this as *Project Tier based Rate Limits* vs *System/Global Endpoint based*, which i would describe as system guardrails that work in tandem with the project tiers, but are global and unless specified in a contract, are the global maximum regardless of project tier.

File: docs/guides/rate-limits-project-based.mdx [hunk 2]
```diff
@@ -0,0 +1,100 @@
+---
+id: project-based-rate-limits
+title: Project-Based Rate Limits
+sidebar_label: Project-Based Rate Limits
+---
+
+## Introduction
+
+This page provides a comprehensive explanation of Ory's Project-Based Rate Limiting. These limits are designed to ensure fair
```
Comment (Demonsthere) [inline docs/guides/rate-limits-project-based.mdx:9]:
```suggestion
This page provides a comprehensive explanation of Ory's Project Tier-Based Rate Limiting. These limits are designed to ensure fair
```

File: docs/guides/rate-limits-project-based.mdx [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+Project-Based Rate Limits are controls applied to your entire Ory project, governing the overall request volumes across various
+API endpoints. These limits vary based on your project's plan (Developer, Production, Growth, or Enterprise) and the specific
+environment (Production, Staging, or Workspace).
+
+## How Project-Based Rate Limits Work
+
+Project-Based Rate Limits analyze incoming requests across your entire project and consider factors such as:
+
+- **Project Plan:** Different plans (Developer, Production, Growth, Enterprise) have different rate limit thresholds.
+- **Environment:** Limits may vary between Production, Staging, and Workspace environments.
```
Comment (Demonsthere) [inline docs/guides/rate-limits-project-based.mdx:23]: This is not fully true, the difference between development and stage is mostly in features, and not rates. We had situations where clients would change their projects from dev to stage in hopes to increase their limits which caused confusion.

File: docs/guides/rate-limits-project-based.mdx [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+### Growth Plan Rate Limits
+
+| Environment    | Path / Bucket              | Burst (rps) | Sustained (rpm) |
+| :------------- | :------------------------- | ----------: | --------------: |
+| **Production** | `/sessions/whoami`         |         800 |           18000 |
+|                | `/admin/oauth2/introspect` |         800 |           18000 |
+|                | `/relation-tuples/check`   |         800 |           18000 |
+|                | `GET /admin/identities`    |          20 |             600 |
+|                | `*`                        |         400 |            9000 |
+| **Staging**    | `/sessions/whoami`         |          10 |              30 |
```
Comment (Demonsthere) [inline docs/guides/rate-limits-project-based.mdx:80]: staging has currently the same rates as dev, therefore i dont think it make sense to repeat that here. Clients already did assume that staging has more rates than developer in the past, so this need to be clarified better

File: docs/guides/rate-limits-project-based.mdx [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+|                | `/relation-tuples/check`   |          10 |             300 |
+|                | `GET /admin/identities`    |           1 |              10 |
+|                | `*`                        |           5 |             150 |
+
+## Important Notes on Project-Based Rate Limits
+
+### Workspace Rate Limit
+
+For all projects in a workspace, the rate limit is the same as for the production environment of your plan. For example, on the
+Production plan, the rate limit would be 80 rps on the `/sessions/whoami` path for all projects combined in the workspace.
```
Comment (Demonsthere) [inline docs/guides/rate-limits-project-based.mdx:91]: Are you sure about that? Clients had a single workspace with both prod and dev projects, and they had ratelimit keys based only on the project tier. Our envoy based config has no information about workspaces, only project data

File: docs/guides/rate-limits-endpoint-based.mdx [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+
+This page provides a comprehensive explanation of Ory's Endpoint-Based Rate Limiting. These limits are designed to fortify the
+security of your Ory project endpoints by mitigating common attack vectors such as brute-force and credential stuffing attempts,
+while allowing for flexibility for high-volume legitimate traffic from trusted sources.
+
+## What are Endpoint-Based Rate Limits?
+
+Endpoint-Based Rate Limits are precise controls applied to individual API endpoints within your Ory projects. Unlike Project-Based
+Rate Limits, which govern overall project request volumes, Endpoint-Based Rate Limits focus on safeguarding specific
+functionalities against abuse.
```
Comment (Demonsthere) [inline docs/guides/rate-limits-endpoint-based.mdx:17]: I would rephrase this as system wide rules, which are menat to protect the system, and are a global limit which unless stated in an enterprise contract, cannot be broken regardless of project tier

**Review by Demonsthere — CHANGES_REQUESTED** [review]

File: docs/guides/rate-limits-project-based.mdx (hunk 3 above)
Comment (vinckr) [reply docs/guides/rate-limits-project-based.mdx:23]:
```suggestion
- **Environment:** Limits are low for Development and Staging environments, and high for Production environments.
```

maybe this? 
what is a workspace env? does that mean the difference of rate limit on workspace/project level?

File: docs/guides/rate-limits-project-based.mdx (hunk 5 above)
Comment (tricky42) [reply docs/guides/rate-limits-project-based.mdx]: I copied the current docs on the project tier-based rate-limits with a todo to check with you and Patrik. I will do so this week.

**Comment by tricky42** [comment]
For convenience: 
* https://docs-14aijk0nn-ory.vercel.app/docs/guides/rate-limits
*  https://docs-14aijk0nn-ory.vercel.app/docs/guides/workspaces

File: docs/guides/rate-limits.mdx [hunk 7]
```diff
@@ -1,74 +1,216 @@
...
+- Whitelisted IPs are subject to the same rule structure but with higher limits.
+- To whitelist IPs, please create a support ticket with Ory.
+- This feature ensures that necessary high-volume traffic from trusted sources is not interrupted while maintaining strong
+  security measures.
 
-## Legacy plans
+### List of Rate Limit Rules
 
-The legacy Essential and Scale plans have the same rate limits as Production and Growth respectively.
+This section will be populated later with detailed documentation of the rules set by Ory, including information on standard limits
```
Comment (Demonsthere) [inline docs/guides/rate-limits.mdx:215]: Dunno how detailed we want/can have this section. We could mostly copy this table here https://www.notion.so/Rate-Limits-a0821f7f807f431880e571cf6eb8fb63#c27c0c7124334b4b89326b3e993f25f4

File: docs/guides/rate-limits.mdx [hunk 8]
```diff
@@ -1,74 +1,216 @@
...
-Each of these policies incorporates two types of rate limits:
+Ory implements two main types of rate limits:
 
-1. The `burst` limit, which regulates the maximum number of requests per second, allowing brief spikes in request volume.
-2. The `sustained` limit, controlling the number of requests over a one-minute period.
+1. [**Project Rate Limits in Workspaces**](#project-rate-limits-in-workspaces): These limits are determined by your Ory
+   subscription plan and project environment. They define the maximum number of requests your projects can make to Ory's APIs
+   within a given timeframe. This ensures fair resource allocation across all Ory projects.
+   1. These limits are applied at the project level, meaning all requests originating from a specific project contribute to the
+      rate limit counter for that project.
```
Comment (Demonsthere) [inline docs/guides/rate-limits.mdx:19]: Maybe we should specify more clearly that the project slug is used as an uid?

**Review by Demonsthere — APPROVED** [review]
looks much better imho

**Review by vinckr — COMMENTED** [review]
nice! 

a diagram that visually explains workspaces/projects/environments would be a very good addition here IMO. 
but ofc not a requirement to get it merged.

File: docs/guides/rate-limits.mdx (hunk 7 above)
Comment (tricky42) [reply docs/guides/rate-limits.mdx:215]: We do this as a follow up pr to get this version out asap.

**Comment by tricky42** [comment]
> nice!
> 
> a diagram that visually explains workspaces/projects/environments would be a very good addition here IMO. but ofc not a requirement to get it merged.

Please check and approve ;) 
https://docs-1imj96867-ory.vercel.app/docs/guides/workspaces

**Review by vinckr — APPROVED** [review]

**Pull request merged**